api/apis/integration/service_binding_test.go

```diff
@@ -0,0 +1,280 @@
+					Lifecycle: workloadsv1alpha1.Lifecycle{
+						Type: "buildpack",
+						Data: workloadsv1alpha1.LifecycleData{Buildpacks: []string{"java"}},
+					},
+				},
+			}
+			Expect(
+				k8sClient.Create(context.Background(), cfApp),
+			).To(Succeed())
+
+			_ = createServiceInstance(context.Background(), k8sClient, serviceInstanceGUID, spaceGUID, "service-instance-name", secretName)
+		})
+
+		When("type is `app`", func() {
+			It("is successful and creates a CFServiceBinding resource", func() {
+				var err error
+				req, err = http.NewRequestWithContext(ctx, "POST", serverURI("/v3/service_credential_bindings"), strings.NewReader(fmt.Sprintf(`{
+			"relationships": {
+				"app": {
+					"data": {
+						"guid": %q
+					}
+				},
+				"service_instance": {
+					"data": {
+						"guid": %q
+					}
+				}
+			},
+			"type": "app"
+		}`, appGUID, serviceInstanceGUID)))
+				Expect(err).NotTo(HaveOccurred())
+
+				req.Header.Add("Content-type", "application/json")
+				router.ServeHTTP(rr, req)
+				Expect(rr.Code).To(Equal(201))
+				Expect(rr.Header().Get("Content-Type")).To(Equal("application/json"))
+
+				var response map[string]interface{}
+				Expect(
+					json.NewDecoder(rr.Body).Decode(&response),
+				).To(Succeed())
+
+				Expect(response).To(HaveKeyWithValue("guid", Not(BeEmpty())))
+				guid := response["guid"].(string)
+				Expect(response).To(HaveKeyWithValue("type", Equal("app")))
+				Expect(response).To(HaveKeyWithValue("name", BeNil()))
+				Expect(response).To(HaveKeyWithValue("created_at", Not(BeEmpty())))
+				Expect(response).To(HaveKeyWithValue("updated_at", Not(BeEmpty())))
+				Expect(response).To(HaveKeyWithValue("last_operation", MatchAllKeys(Keys{
+					"description": BeNil(),
+					"state":       Equal("succeeded"),
+					"type":        Equal("create"),
+					"created_at":  Not(BeEmpty()),
+					"updated_at":  Not(BeEmpty()),
+				})))
+				Expect(response).To(HaveKeyWithValue("relationships", map[string]interface{}{
+					"app": map[string]interface{}{
+						"data": map[string]interface{}{
+							"guid": appGUID,
+						},
+					},
+					"service_instance": map[string]interface{}{
+						"data": map[string]interface{}{
+							"guid": serviceInstanceGUID,
+						},
+					},
+				}))
+				Expect(response).To(HaveKeyWithValue("metadata", map[string]interface{}{
+					"annotations": map[string]interface{}{},
+					"labels":      map[string]interface{}{},
+				}))
+				Expect(response).To(HaveKeyWithValue("links", MatchAllKeys(Keys{
+					"app":              HaveKeyWithValue("href", HaveSuffix("/v3/apps/"+appGUID)),
+					"service_instance": HaveKeyWithValue("href", HaveSuffix("/v3/service_instances/"+serviceInstanceGUID)),
+					"self":             HaveKeyWithValue("href", HaveSuffix("/v3/service_credential_bindings/"+guid)),
+					"details":          HaveKeyWithValue("href", HaveSuffix("/v3/service_credential_bindings/"+guid+"/details")),
+				})))
+
+				serviceBinding := new(servicesv1alpha1.CFServiceBinding)
```

Comment:
The test so far has been checking the response. From here on the test checks the k8s object. Therefore, that should be a separate `It`. Also, if creating the object is performed in a `JustBeforeEach`, that It would be just a sibling to the response checking one.